src/server/game/Globals/WorldGlobals.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * This file is part of the AzerothCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by the
+ * Free Software Foundation; either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "GameTime.h"
+#include "Log.h"
+#include "QueryResult.h"
+#include "WorldGlobals.h"
+
+WorldGlobals* WorldGlobals::instance()
+{
+    static WorldGlobals instance;
+    return &instance;
+}
+
+void WorldGlobals::LoadAntiDosOpcodePolicies()
+{
+    uint32 oldMSTime = getMSTime();
+
+    QueryResult result = WorldDatabase.Query("SELECT Opcode, Policy, MaxAllowedCount FROM antidos_opcode_policies");
+    if (!result)
+    {
+        LOG_WARN("server.loading", ">> Loaded 0 AntiDos Opcode Policies. DB table `antidos_opcode_policies` is empty!");
+        LOG_INFO("server.loading", " ");
+        return;
+    }
+
+    uint32 count = 0;
+
+    do
+    {
+        Field* fields = result->Fetch();
+
+        uint16 opcode = fields[0].Get<uint16>();
+        if (opcode >= NUM_OPCODE_HANDLERS)
+            continue;
+
+        std::unique_ptr<AntiDosOpcodePolicy> policy = std::make_unique<AntiDosOpcodePolicy>();
+        policy->Policy = fields[1].Get<uint8>();
+        policy->MaxAllowedCount = fields[2].Get<uint16>();
+
+        _antiDosOpcodePolicies[opcode] = std::move(policy);
+
+        ++count;
+    } while (result->NextRow());
+
+    LOG_INFO("server.loading", ">> Loaded {} AntiDos Opcode Policies in {} ms", count, GetMSTimeDiffToNow(oldMSTime));
```

Comment:
Could use _antiDosOpcodePolicies.size() instead of count.

Also for reload case we need to clear _antiDosOpcodePolicies before loading into it?

Reply:
It's an array for O(1) lookup speed, size would return the full length of the array rather than the amount of non-null elements.

But yes, you're right with needing to clear it (or in this case, default nullptr init the full array)

Reply:
Alright!